src/GafferSceneModule/VisibleSetBinding.cpp

```diff
@@ -60,14 +60,28 @@ void GafferSceneModule::bindVisibleSet()
 
 	IECorePython::TypedDataFromType<VisibleSetData>();
 
-	class_<VisibleSet>( "VisibleSet" )
+	scope s = class_<VisibleSet>( "VisibleSet" )
 		.def( init<>() )
 		.def( init<const VisibleSet &>() )
-		.def( "match", (PathMatcher::Result (VisibleSet ::*)( const std::vector<InternedString> &, const size_t ) const)&VisibleSet::match, arg( "minimumExpansionDepth" ) = 0 )
+		.def( "match", (VisibleSet::Result (VisibleSet ::*)( const std::vector<InternedString> &, const size_t ) const)&VisibleSet::match, arg( "minimumExpansionDepth" ) = 0 )
 		.def_readwrite( "expansions", &VisibleSet::expansions )
 		.def_readwrite( "inclusions", &VisibleSet::inclusions )
 		.def_readwrite( "exclusions", &VisibleSet::exclusions )
 		.def( "__eq__", &VisibleSet::operator== )
 	;
 
+	scope r = class_<VisibleSet::Result>( "Result" )
+		.def( init<>() )
+		.def( init<VisibleSet::Result::DrawMode, bool>() )
+		.def_readwrite( "descendantsVisible", &VisibleSet::Result::descendantsVisible )
+		.def_readwrite( "drawMode", &VisibleSet::Result::drawMode )
+		.def( "__eq__", &VisibleSet::Result::operator== )
```

Comment:
This works, but I think the official boosty way is `def( self == self )`.

Reply:
Would be nice to define `__repr__` perhaps - would make it easier to introspect with VisibleSet in the PythonEditor and suchlike.

Reply:
Ah, good to know! I've taken the boosty way in 2e467b8.

Adding `__repr__` is a good idea, I've done that in f763d6b, though please let me know if there's a more straightforward way.

Reply:
There is a slightly more minimal way of doing the repr if you're interested - you can rely on the fact that repr is already defined for the DrawMode enum : https://github.com/johnhaddon/gaffer/commit/b1b974f1e839ce4cce790ad45fc28f2f38231a35

Reply:
Neat! I've taken that approach in the now squashed down 2ac3b58.